TenantFile/Api/Models/TenantContext.cs

```diff
@@ -0,0 +1,55 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Npgsql.EntityFrameworkCore;
+
+namespace TenantFile.Api.Models
+{
+    // https://dev.to/michaelstaib/get-started-with-hot-chocolate-and-entity-framework-e9i
+    public class TenantContext : DbContext
+    {
+        private string _connectionString;
+        public TenantContext(IConfiguration configuration)
+        {
+            _connectionString = configuration.GetValue<string>("DbUrl");
+        }
+
+        public DbSet<Tenant> Tenants { get; set; } = null!;
+        public DbSet<Property> Properties { get; set; } = null!;
+        public DbSet<Phone> Phones { get; set; } = null!;
+
+        protected override void OnConfiguring(DbContextOptionsBuilder options)
+        {
+            options.UseNpgsql(_connectionString);
+        }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
```

Comment:
We should move the configurations to their own classes like:

```
public class TenantPhoneConfiguration : IEntityTypeConfiguration<TenantPhone>
{
        public void Configure(EntityTypeBuilder<TenantPhone> modelBuilder)
{
 modelBuilder.Entity<TenantPhone>()
            .HasKey(tp => new { tp.TenantId, tp.PhoneId });
            modelBuilder.Entity<TenantPhone>()
           
            .HasOne(tp => tp.Tenant)
            .WithMany(t => t.TenantPhones)
            .HasForeignKey(t => t.TenantId); 
```

            modelBuilder.Entity<TenantPhone>()
                .HasOne(tp => tp.Phone)
                .WithMany(p => p.TenantPhones)
                .HasForeignKey(tp => tp.PhoneId);
}
and call:

`modelBuilder.ApplyConfigurationsFromAssembly(typeof(TenantContext).Assembly);`

in the OnModelCreating method so the configurations are easy to find and modify as we add for entities.

Reply:
I agree, I'll work on a re-spin that refactors this.